[tool.poetry]
name = "mano-auth-server"
version = "0.1.0"
description = ""
authors = ["vunk.lai <vunk.lai@zyxel.com.tw>"]
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.9"
# django
Django = "^4.1.3"
djangorestframework = "^3.14.0"
django-cors-headers = "^3.13.0"
django-extensions = "^3.2.1"
# other
PyJWT = {extras = ["crypto"], version = "^2.6.0"}

[tool.poetry.group.dev.dependencies]
ipython = "^8.6.0"
# format
isort = "^5.10.1"
black = "^22.10.0"
# linter
# pylint 2.15 causes django.core.exceptions.Improperlyconfigured to be raised
# https://github.com/PyCQA/pylint-django/issues/370
pylint = "2.14.5" # "^2.15.2"
pylint-django = "^2.5.3"
mypy = "^0.982"
# analysis
bandit = "^1.7.4"
safety = "^2.3.1"
# test
pytest = "^7.2.0"
pytest-django = "^4.5.2"

[tool.isort]
profile = "black"
known_django = "django"
sections = "FUTURE,STDLIB,DJANGO,THIRDPARTY,FIRSTPARTY,LOCALFOLDER"
skip = "migrations"

[tool.pylint.main]
ignore = 'db.sqlite3'
ignore-patterns = ['migrations/', 'private.key']
load-plugins = ['pylint_django']
django-settings-module = 'server.settings'

[tool.pylint.messages_control]
max-line-length = 120
disable = [
  'missing-module-docstring',
  'missing-class-docstring',
  'missing-function-docstring',
  'too-few-public-methods',
  'too-many-instance-attributes',
  'too-many-locals',
  'no-member',
  'fixme',
  'invalid-name',
  'imported-auth-user',
]

[tool.bandit]
skips = [
  'B101', # fixme: assert_used is not working on toml settings
  'B105', # fixme: use django-environ
 ]

[tool.pytest.ini_options]
DJANGO_SETTINGS_MODULE = "server.settings"
python_files = ['tests.py', 'test_*.py']

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
